<launch>
    <arg name="type" /> <!-- concurrent or sequential -->
    <arg name="scene" default="toolbox"/>
    <arg name="ip" /> <!-- IP of the VRPN server -->
    <arg name="port" default="3883" /> <!-- Port of the VRPN server (default 3883) -->
    <arg name="policy" default="hardcoded" /> <!-- Policy values: hardcoded, wait, normal -->
    <arg name="name" default="none" /> <!-- Name of log files -->
    <arg name="display" default="debug" /> <!-- debug or face display -->

    <rosparam param="/thr/logs_name" subst_value="True">$(arg name)</rosparam>
    <rosparam param="/thr/display" subst_value="True">$(arg display)</rosparam>

    <include file="$(find thr_coop_assembly)/launch/perception.launch" >
      <arg name="ip" value="$(arg ip)"/> <!-- IP of the VRPN server (like Motive or Arena) -->
      <arg name="port" value="$(arg port)"/> <!-- Port of the VRPN server (default 3883) -->
      <arg name="scene" value="$(arg scene)"/>
	  <arg name="type" value="$(arg type)"/>
    </include>
       
    <include file="$(find thr_coop_assembly)/launch/action_server_$(arg type).launch" />

    <!--<node pkg="google_tts_ros" name="text_to_speech" type="say.py" output="log"/>-->
    <node pkg="thr_coop_assembly" name="$(arg type)_$(arg display)_display" type="$(arg type)_$(arg display)_display.py" output="log"/>
    <node pkg="thr_coop_assembly" name="keyboard_interaction_controller_$(arg type)" type="keyboard_interaction_controller_$(arg type).py" output="screen"/>
</launch>
